apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/usersettings/networkAndServers/NetworkAndServers.kt

```diff
@@ -50,21 +50,34 @@ fun ModalData.NetworkAndServersView(close: () -> Unit) {
   val currUserServers = remember { stateGetOrPut("currUserServers") { emptyList<UserOperatorServers>() } }
   val userServers = remember { stateGetOrPut("userServers") { emptyList<UserOperatorServers>() } }
   val serverErrors = remember { stateGetOrPut("serverErrors") { emptyList<UserServersError>() } }
-  val scope = rememberCoroutineScope()
 
   val proxyPort = remember { derivedStateOf { appPrefs.networkProxy.state.value.port } }
-  ModalView(
-    close = {
-      if (!serversCanBeSaved(currUserServers.value, userServers.value, serverErrors.value)) {
-        close()
-      } else {
-        showUnsavedChangesAlert(
-          { scope.launch { saveServers(currentRemoteHost?.remoteHostId, currUserServers, userServers) }},
-          close
-        )
+  fun onClose(): Boolean = if (!serversCanBeSaved(currUserServers.value, userServers.value, serverErrors.value)) {
+    chatModel.centerPanelBackgroundClickHandler = null
+    close()
+    false
+  } else {
+    showUnsavedChangesAlert(
+      {
+        CoroutineScope(Dispatchers.Default).launch {
+          saveServers(currentRemoteHost?.remoteHostId, currUserServers, userServers)
+          chatModel.centerPanelBackgroundClickHandler = null
+          ModalManager.start.closeModals()
+        }
+      },
+      {
+        chatModel.centerPanelBackgroundClickHandler = null
```

Comment:
easy to forget `chatModel.centerPanelBackgroundClickHandler = null` and leave it model, maybe always set it to null in `clickable(...` in App?